app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -538,12 +539,22 @@ class LuciBuildService {
     log.info('Task commit_key: $commitKey for task name: ${task.name}');
     log.info('Task task_key: $taskKey for task name: ${task.name}');
 
+    // Received by [PostsubmitLuciSubscription] when LUCI phones back with build results.
     final Map<String, dynamic> rawUserData = <String, dynamic>{
       'commit_key': commitKey,
       'task_key': taskKey,
     };
 
-    await createPostsubmitCheckRun(commit, target, rawUserData);
+    final checkRun = existingCheckRun ?? await createPostsubmitCheckRun(commit, target);
```

Comment:
Maybe the wording is getting me confused but it looks like you are finding a checkrun but ultimately creating a new one again?

Reply:
So, my goal here was to use the `??` null operator to create a new check run if `existingCheckRun` was not passed or if it was explicitly passed as null. This is because this `_createPostsubmitScheduleBuild()` hosting method is also used for brand new builds.

Maybe I could think of a way to clarify it.

---

There is an edge case where maybe the `existingCheckRun` is not found way up in the `ResetProdTask` request handler:
https://github.com/flutter/cocoon/pull/2343/files#diff-8b18660d0517d62995b30d9277a6cd6601f428ccd5def5d4535f668cd6ba8cb3R90

But to make that less likely, I have `listCheckRunsForRef` doing retries so we can at least mostly factor out HTTP connection errors:
https://github.com/flutter/cocoon/pull/2343/files#diff-fd1f79cec05184daf8caf11256548ca4b3ff25cc840d6cf14f9f0cfc6c31d0c6R147